sotodlib/coords/optics.py

```diff
@@ -655,3 +562,147 @@ def SAT_focal_plane(aman, x=None, y=None, pol=None, rot=0, mapping_data=None):
             aman.wrap("focal_plane", focal_plane)
 
     return xi, eta, gamma
+
+
+def get_focal_plane(
+    aman,
+    x=None,
+    y=None,
+    pol=None,
+    roll=0,
+    telescope_flavor=None,
+    tube_slot=None,
+    wafer_slot="ws0",
+    config_path=None,
+    zemax_path=None,
+    mapping_data=None,
+    return_fp=False,
+):
+    """
+    Unified interface for LAT_focal_plane and SAT_focal_plane including the step of applying ufm_to_fp.
+
+    Arguments:
+
+        aman: AxisManager nominally containing aman.det_info.wafer.det_x and aman.det_info.wafer.det_y.
+              If provided focal plane will be stored in aman.focal_plane.
+
+        x: Detector x positions, if provided will override positions loaded from aman.
+
+        y: Detector y positions, if provided will override positions loaded from aman.
+
+        pol: Detector polarization angle, if provided will override positions loaded from aman.
+
+        roll: Rotation about the line of sight.
+              For the LAT this is elev - 60 - corotator.
+              For the SAT this is -1*boresight.
+
+        telescope_flavor: What the telescope flavor is ('LAT' or 'SAT').
+                          Note that this is case insenstive.
+                          If None and aman contains obs_info, will be extracted from there.
+
+        tube_slot: Which optics tube of the telescope to use.
+                   Only used by the LAT.
+                   If None and aman contains obs_info, will be extracted from there.
+
+        wafer_slot: Which wafer slot to use.
+
+        config_path: Path to the ufm_to_fp config file.
+
+        zemax_path: Path to the data file from Zemax.
+                    Only used by the LAT.
+
+        mapping_data: Tuple of (x, theta) that can be interpolated to map the focal plane to the sky.
+                      Leave as None to use the default mapping.
+                      Only used by the SAT.
+
+        return_fp: If True also return the transformed focal plane in mm.
+
+    Returns:
+
+        xi: Detector elev on sky from physical optics in radians.
+            If aman is provided then will be wrapped as aman.focal_plane.xi.
+
+        eta: Detector xel on sky from physical optics in radians.

```

Comment:
"xel"?  I see "elev and xel" used all over this module... I don't know what that means, but it is unlikely to be what "xi" and "eta" usually mean.

Reply:
Sorry left over from the Simon script I adapted this from. They are the same (modulo the zemax rotation) its just radio astronomy lingo I think. Will swap them out.